### ISSUE TITLE
How to monitor a validation loss or metric?

### ISSUE THREAD

**Issue by gustavu92** [yassouali/pytorch_segmentation#25]
How can I modify config.json for monitor a validation loss or metric in the earlystop in training process?

**Comment by yassouali** [comment]
you can monitor any metric in the returned dict `log` in the `_valid_epoch`. For instance to monitor the validation loss, try replacing `"monitor": "max Mean_IoU"` with `"monitor": "min val_loss"`

**Comment by gustavu92** [comment]
I think it is useful to change the validation metrics name to add some prefix like: "val_" . Maybe adding **self.wrt_mode** before the metric name in **_get_seg_metrics** funtion. Example: self.wrt_mode + "_Pixel_Accuracy". Because if we set the checkpoint monitor for "max Pixel_Accuracy" it will save the best model also using the training metric.

**Comment by yassouali** [comment]
Only the validation metrics are tracked, the checking is executed only after an eval was conducted.
